godot-core/src/obj/script.rs

```diff
@@ -153,17 +153,26 @@ pub trait ScriptInstance: Sized {
     /// The engine may call this function if
     /// [`IScriptExtension::is_placeholder_fallback_enabled`](crate::classes::IScriptExtension::is_placeholder_fallback_enabled) is enabled.
     fn property_set_fallback(this: SiMut<Self>, name: StringName, value: &Variant) -> bool;
+
+    #[cfg(since_api = "4.3")]
+    fn get_method_argument_count(&self, _method: StringName) -> Option<u32> {
+        None
+    }
```

Comment:
> Should this have a default implementation, or would it be better to require implementing it?

the engine will ignore the return value if `r_is_valid` is `false` and use a fallback value. To avoid breaking existing user code I choose to make the trait method optional as well. Otherwise existing code will stop compiling once someone upgrades to 4.3.

> In the future, would of course be nice to somehow automate this

We currently do not store any state for script instances. At the moment everything is more or less a pass-through with a nicer API. 

> so that the user doesn't have to maintain a match statement with function strings, which is very error-prone...

That would indeed be a quite curde implementation. I imagine, most script implementations would use at least a HashMap or similar to look up the methods on a script and fetch the argument count from some meta data of the script runtime.

Reply:
> To avoid breaking existing user code I choose to make the trait method optional as well. Otherwise existing code will stop compiling once someone upgrades to 4.3.

But apart from breaking existing code, which choice do you think is better in terms of ergonomics? Isn't there a risk to forget overriding this, or what would be the consequences?

Reply:
> which choice do you think is better in terms of ergonomics?

Making it required would be definitely less error-prone. Disregarding backwards compatibility, I would prefer that.

>  Isn't there a risk to forget overriding this, or what would be the consequences?

If the method is not implemented, public interfaces like `Callable::get_argument_count` and `Object::get_method_argument_count` will return `0`.

Reply:
I'd say let's go for it then :+1: 
v0.2 is a chance to break some code in cases where it improves the overall experience.

Reply:
Btw, do you know how the semantics for argument counts are before Godot 4.3?
If someone targets an older API version, what happens?

Reply:
As far as I know, argument counts weren't a thing before 4.3. It was introduced in this PR https://github.com/godotengine/godot/pull/87680

Reply:
I see, thank you!